go-controller/pkg/ovn/master.go

```diff
@@ -888,6 +934,12 @@ func (oc *Controller) allocateNodeSubnets(node *kapi.Node) ([]*net.IPNet, []*net
 	// assume IP families match, i.e. no IPv6 config and node annotation IPv4
 	if expectedHostSubnets == currentHostSubnets {
 		klog.Infof("Allocated Subnets %v on Node %s", hostSubnets, node.Name)
+		// Ensure that the parsed host subnets are reserved in the allocator
+		for _, subnet := range hostSubnets {
+			//err = oc.masterSubnetAllocator.MarkAllocatedNetwork(subnet)
```

Comment:
are lines 937-942 necessary?

Reply:
so that's something I think we are missing. When the subnet annotations are already existing and match the expected subnets count, we don't seem to mark them as allocated today.

Reply:
I guess what I'm not following is what the subnet annotations have to do with your patch. Is this something separate that you just noticed while you were fixing this other stuff? Should it be a different PR?

Reply:
Something separate I happened to notice. Those changes are commented out precisely so we can have a discussion. If we think that is indeed the case that the subnets are not marked allocated when the expected count matches the annotations count, then I can fix it in a separate PR. Likewise with the fix wrt `defer()`